### PR TITLE
Immu refactor

### PR DESCRIPTION
**Pull request by wy65701436**

### CODE REVIEW

File: src/controller/immutable/controller.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"fmt"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/immutable"
+
+	"github.com/goharbor/harbor/src/pkg/immutable/model"
+)
+
+var (
+	// Ctr is a global variable for the default immutable controller implementation
+	Ctr = NewAPIController(immutable.NewDefaultRuleManager())
```
Comment (heww) [inline src/controller/immutable/controller.go:14]: Rename `Ctr` to `Ctl` when this controller is new.

File: src/pkg/retention/job.go [hunk 2]
```diff
@@ -111,7 +111,7 @@ func (pj *Job) Run(ctx job.Context, params job.Parameters) error {
 	}
 
 	// Run the flow
-	results, err := processor.Process(allCandidates)
+	results, err := processor.Process(ctx.SystemContext(), allCandidates)
```
Comment (ywk253100) [inline src/pkg/retention/job.go:114]: Is the `ctx.SystemContext()` with a `ormer` pupoluated? Seems the `ctx` is passed to a DAO method in the final

**Review by heww — APPROVED** [review]

File: src/pkg/immutable/dao/dao.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	ListImmutableRules(ctx context.Context, query *q.Query) ([]*model.ImmutableRule, error)
+	DeleteImmutableRule(ctx context.Context, id int64) error
+}
+
+// New creates a default implementation for DAO
+func New() DAO {
+	return &iDao{}
+}
+
+type iDao struct{}
```
Comment (heww) [inline src/pkg/immutable/dao/dao.go:28]: rename `iDao` to `dao` like other pkgs?

File: src/pkg/retention/job.go (hunk 2 above)
Comment (heww) [reply src/pkg/retention/job.go:114]: yup, the ctx.SystemContext() returns a context.Context with a ormer pupoluated.

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**